### PR TITLE
PARQUET-1035 Write Int96 from Arrow timestamp(ns)

### PR DESCRIPTION
**Pull request by c-nichols**

### CODE REVIEW

File: src/parquet/arrow/writer.cc [hunk 1]
```diff
@@ -51,6 +51,22 @@ using parquet::schema::GroupNode;
 namespace parquet {
 namespace arrow {
 
+constexpr int64_t JULIAN_EPOCH_OFFSET_DAYS = 2440588LL;
```
Comment (c-nichols) [inline src/parquet/arrow/writer.cc]: Google C++ style guide requires constants to be named like kJulianEpochOffsetDays

File: src/parquet/arrow/writer.cc [hunk 2]
```diff
@@ -51,6 +51,22 @@ using parquet::schema::GroupNode;
...
+constexpr int64_t NANOSECONDS_PER_DAY = 86400LL * 1000LL * 1000LL * 1000LL;
+
+static inline Status nanoseconds_to_impala_timestamp(const int64_t nanoseconds, Int96* impala_timestamp) {
```
Comment (c-nichols) [inline src/parquet/arrow/writer.cc]: need mixed case here accordin gto google style guide

File: src/parquet/arrow/writer.cc [hunk 3]
```diff
@@ -462,6 +498,33 @@ Status FileWriter::Impl::WriteNullableBatch<Int32Type, ::arrow::Time32Type>(
...
+  if (type.unit() == TimeUnit::NANO) {
+    for (int i = 0; i < num_values; i++) {
+      if (bitset_valid_bits & (1 << bit_offset_valid_bits)) {
+        PARQUET_CATCH_NOT_OK(
+            nanoseconds_to_impala_timestamp(data_ptr[i], buffer_ptr + i));
+      }
+      READ_NEXT_BITSET(valid_bits);
+    }
+  } else {
+    // not implement it
```
Comment (csdev) [inline src/parquet/arrow/writer.cc]: `Status::NotImplemented`

File: src/parquet/arrow/writer.cc (hunk 2 above)
Comment (csdev) [inline src/parquet/arrow/writer.cc]: unit test?

**Review by csdev — CHANGES_REQUESTED** [review]

**Review by csdev — APPROVED** [review]

**Pull request closed without merging**